platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerImpl.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.io.filesystem.internal;
+
+import static com.swirlds.common.io.utility.FileUtils.rethrowIO;
+import static com.swirlds.logging.legacy.LogMarker.STARTUP;
+import static java.nio.file.Files.exists;
+
+import com.swirlds.common.io.filesystem.FileSystemManager;
+import com.swirlds.common.io.utility.FileUtils;
+import com.swirlds.common.io.utility.RecycleBin;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileAttribute;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Default implementation of {@link FileSystemManager} It organizes the file creation in the following structure: root
+ * usr tmp recycle-bin
+ * <p>
+ * If root directory already exist, deletes its content and recreates it. All {@link Path}s provided by this class are
+ * handled in the same filesystem as indicated in rootLocation parameter.
+ *
+ * @implNote Creates and manages a temp dir when creating an instance of this class using
+ * {@link Files#createTempDirectory(Path, String, FileAttribute[])} located under {@code rootPath} where all temporary
+ * files are located.
+ */
+public class FileSystemManagerImpl implements FileSystemManager {
+
+    private static final Logger logger = LogManager.getLogger(FileSystemManagerImpl.class);
+    private static final String TMP = "tmp";
+    private static final String USER = "usr";
+    private static final String BIN = "recycle-bin";
+    private final Path rootPath;
+    private final Path tempPath;
+    private final Path userPath;
+    private final Path recycleBinPath;
+    private final RecycleBin bin;
+    private static final AtomicLong TMP_FIELD_INDEX = new AtomicLong(0);
```

Comment:
I don't think this should be static. We shouldn't try to prevent collisions between two file system managers. Instead, we should rely on the fact that there should only be a single file system manager that has ownership of a directory.

Reply:
Okay, but nothing prevents multiple instances of the fileSystemManager from being created at the same location. If the index is not static, those two instances could create files in the same directory with the same name. Thoughts?

Reply:
I suggest we get a small group of decision makers together to discuss static vs non-static behavior. My personal preference is to take steps towards eliminating static stateful variables entirely.